mypyc/irbuild/specialize.py

```diff
@@ -143,6 +143,9 @@ def translate_safe_generator_call(
                 val = tuple_from_generator_helper(builder, expr.args[0])
                 if val is not None:
                     return val
+            if callee.fullname == "builtins.set":
+                return translate_set_comprehension(builder, expr.args[0])
+
```

Comment:
Maybe I should move `builtins.set` to a new specialize function ?

Reply:
That sounds reasonable, since the implementation isn't sharing much with the other cases.